src/haddock/modules/analysis/contactmap/contmap.py

```diff
@@ -0,0 +1,1843 @@
+    """
+    try:
+        sorted_models = sorted(models, key=lambda m: m.score)
+    except AttributeError:
+        sorted_models = models
+    finally:
+        subset_bests = sorted_models[:topX]
+    return subset_bests
+
+
+####################
+# Define functions #
+####################
+def extract_pdb_dt(path: Path) -> dict:
+    """Read and extract ATOM/HETATM records from a pdb file.
+
+    Parameters
+    ----------
+    path : Path
+        Path to a pdb file.
+
+    Return
+    ------
+    pdb_chains : dict
+        A dictionary of the pdb file accesible using chains as keys.
+    """
+    pdb_chains: dict = {'chain_order': []}
+    # Read file
+    with open(path, 'r') as f:
+        # Loop over lines
+        for _ in f:
+            # Skip non ATOM / HETATM lines
+            if not any([
+               _.startswith('ATOM'),
+               _.startswith('HETATM'),
+               ]):
+                continue
+
+            # Extract residue name
+            resname = _[slc_resname]
+            # Extract chain id
+            chainid = _[slc_chainid]
+            # Extract resid
+            resid = _[slc_resseq].strip()
+
+            # Check if chain already parsed
+            if chainid not in pdb_chains.keys():
+                # Add to ordered chains
+                pdb_chains['chain_order'].append(chainid)
+                # Initiate new chain holder
+                pdb_chains[chainid] = {'order': []}
+
+            # Check if new resid id
+            if resid not in pdb_chains[chainid].keys():
+                # Add to oredered resids
+                pdb_chains[chainid]['order'].append(resid)
+                # Initiate new residue holder
+                pdb_chains[chainid][resid] = {
+                    'index': len(pdb_chains[chainid]['order']) - 1,
+                    'resname': resname,
+                    'chainid': chainid,
+                    'resid': resid,
+                    'position': len(pdb_chains[chainid]['order']),
+                    'atoms_order': [],
+                    'atoms': {},
+                    }
+            # extract atome name
+            atname = _[slc_name].strip()
+            # check if not an hydrogen
+            if atname[0] == 'H':
+                continue
+
+            # extact atome coordinates
+            coords = extract_pdb_coords(_)
+            pdb_chains[chainid][resid]['atoms_order'].append(atname)
+            pdb_chains[chainid][resid]['atoms'][atname] = coords
+
+    return pdb_chains
+
+
+def extract_pdb_coords(line: str) -> list[float]:
+    """Extract coordinated from a PDB line.
+
+    Parameters
+    ----------
+    line : str
+        A strandard ATOM/HETATM pdb record.
+
+    Return
+    ------
+    coords : list[float]
+        List of the X, Y and Z coordinate of this atom.
+    """
+    x = float(line[slc_x].strip())
+    y = float(line[slc_y].strip())
+    z = float(line[slc_z].strip())
+    coords = [x, y, z]
+    return coords
```

Comment:
don't we already have something in https://github.com/haddocking/haddock3/blob/main/src/haddock/libs/libalign.py for this?

Reply:
Similar but not equal, unfortunately